### PR TITLE
fix: introduce asynchronous, revision-aware config snapshot saving to keep event loops responsive and prevent stale writes from overwriting newer state

### LINKED ISSUE #9301: [Bug] Weixin OC 账号状态同步保存会阻塞 asyncio 事件循环
## What happened / 发生了什么

在群晖 NAS 的生产环境中，AstrBot 偶发出现 WebUI 端口请求超时，Docker 健康检查随后可能将容器标记为 `unhealthy`。Python 主进程仍然存活，磁盘操作返回后服务会自行恢复。

事件循环监视日志记录到一次约 43 秒的停顿。脱敏后的调用栈显示，Weixin OC 轮询任务在更新 `get_updates_buf` 后保存账号状态，并在 asyncio 主线程中同步执行 `astrbot_config.save_config()`：

```text
Event loop lag detected: 43.717s (threshold 15.000s)

astrbot/core/config/astrbot_config.py:244 save_config
astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py:588 _save_account_state
astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py:1612 _poll_inbound_updates
astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py:1759 run
```

`save_config()` 会依次执行临时 JSON 文件写入、`flush()`、`os.fsync()` 和 `os.replace()`。当 NAS 存储响应时间上升时，同步 `fsync` 会阻塞整个事件循环。在此期间，WebUI、OneBot 以及同一事件循环中的其他任务均无法及时响应。

相关内网地址、账号标识、令牌、消息内容和持久目录名称均已省略。

## Reproduce / 如何重现

1. 启用 Weixin OC 平台适配器。
2. 让入站轮询收到会改变同步缓冲区的更新。
3. 使配置文件所在存储产生较高的写入或 `fsync` 延迟，例如繁忙的 NAS btrfs 卷。
4. 观察 `_poll_inbound_updates()` 调用 `_save_account_state()`。
5. 此时事件循环延迟监视会记录明显停顿，HTTP 健康检查可能在超时窗口内失败。

常规低延迟磁盘环境中较难稳定触发；NAS 写入延迟会放大该问题。

## Environment / 环境

- AstrBot: v4.26.6
- Deployment: Docker Compose
- Host: Synology NAS，DSM Container Manager
- Docker Engine: 24.0.2
- Storage driver / filesystem: btrfs
- OS: Linux
- Messaging platform: Weixin OC
- WebUI health probe timeout: 3 seconds

## Expected behavior / 预期行为

账号状态仍应持久化，但文件写入和 `fsync` 不应占用 asyncio 主事件循环。较慢的 NAS 存储只应延长保存任务本身，不应同时阻塞 WebUI、消息平台和其他协程。

## Proposed fix / 修正方案

将 `astrbot_config.save_config` 交给 `asyncio.to_thread()` 执行，并增加定向测试确认保存操作在线程中运行。

对应 PR：#9300

https://github.com/AstrBotDevs/AstrBot/pull/9300

定向验证：

```text
python -m pytest -q tests/test_weixin_oc_state_persistence.py tests/test_weixin_oc_login_registration.py
4 passed
```

PR 的格式检查、完整 pytest、Dashboard 构建、CodeQL，以及 Windows、macOS、Ubuntu 上 Python 3.10 至 3.14 的 Smoke Test 均已通过。

## Code of Conduct

- [x] 已阅读并同意遵守项目 Code of Conduct。

### PR DESCRIPTION
**Pull request by RhoninSeiei**
### 改动

- 为 `AstrBotConfig` 增加稳定快照异步保存接口；共享配置的深拷贝在事件循环中完成，`json.dump`、`flush`、`fsync` 与原子替换在线程中执行。
- 使用保存修订号丢弃乱序完成的旧快照，避免旧状态覆盖新配置；快照锁与提交锁分离，慢速 NAS 元数据操作不会阻塞下一次快照生成。
- Weixin OC 账号状态改用异步保存，并以提交结果和 context token 修订号判断是否清除 dirty 标记。

Fixes #9301

### 验证

```text
python -m pytest -q tests/test_weixin_oc_state_persistence.py tests/test_weixin_oc_login_registration.py tests/unit/test_config.py
51 passed

ruff format --check astrbot/core/config/astrbot_config.py astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py tests/test_weixin_oc_state_persistence.py tests/unit/test_config.py
4 files already formatted

ruff check astrbot/core/config/astrbot_config.py astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py tests/test_weixin_oc_state_persistence.py tests/unit/test_config.py
All checks passed!
```

测试覆盖稳定快照、事件循环响应、慢 `fsync`、慢 `os.replace`、乱序保存、旧快照丢弃，以及 context token 的 A→B 和 A→B→A 时序。

- [x] This is NOT a breaking change. / 这不是一个破坏性变更。
- [x] 未引入新依赖。

## Summary by Sourcery

Introduce asynchronous, revision-aware config snapshot saving to keep event loops responsive and prevent stale writes from overwriting newer state.

New Features:
- Add an asynchronous stable snapshot save API for AstrBotConfig that offloads blocking file operations to a background thread.
- Make Weixin OC account state persistence use the asynchronous config save API with revision tracking for context tokens.

Bug Fixes:
- Prevent slow fsync/replace operations from blocking the event loop or causing older snapshots to overwrite newer configuration.
- Ensure Weixin OC context token dirty flags are only cleared when the corresponding async save commits the latest revision, avoiding loss of newer state.

Tests:
- Add unit tests covering async config persistence behavior under slow fsync/replace, snapshot stability, and out-of-order commits.
- Add tests verifying Weixin OC account state uses async persistence and correctly handles context token revision and dirty-flag edge cases.

### CODE REVIEW

File: astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ async def _save_account_state(self) -> None:
             break
 
         self._sync_client_state()
-        astrbot_config.save_config()
+        await asyncio.to_thread(astrbot_config.save_config)
```
[the remark this answers is not in the document]
Comment (RhoninSeiei) [reply astrbot/core/platform/sources/weixin_oc/weixin_oc_adapter.py]: 已在 `1fcc22ffa` 处理该并发问题：

1. `AstrBotConfig.save_config_async()` 在事件循环中先生成深拷贝快照，工作线程只负责序列化该快照以及 `flush/fsync/os.replace`，不会再遍历共享配置对象。
2. 为保存分配单调修订号；乱序完成的旧快照不会覆盖较新的配置文件，异步接口会返回该快照是否实际提交。
3. 快照锁与提交锁分离，NAS 上较慢的 `os.replace()` 不会使下一次异步保存阻塞事件循环。
4. Weixin 使用 context token 修订号与提交结果判断是否清除 dirty 标记，覆盖 A→B 和 A→B→A 时序。

新增测试覆盖事件循环响应、稳定快照、乱序保存、慢 `os.replace`、旧快照丢弃及 token ABA。定向测试共 `51 passed`，Ruff 检查与格式检查通过。

File: astrbot/core/config/astrbot_config.py [hunk 2]
```diff
@@ -49,6 +52,9 @@ def __init__(
         object.__setattr__(self, "config_path", config_path)
         object.__setattr__(self, "default_config", default_config)
         object.__setattr__(self, "schema", schema)
+        object.__setattr__(self, "_save_state_lock", threading.Lock())
```
[the remark this answers is not in the document]
Comment (RhoninSeiei) [reply astrbot/core/config/astrbot_config.py:55]: 这里保留两把锁是为了避免重新引入事件循环阻塞。

`_prepare_config_snapshot()` 在事件循环线程中同步获取 `_save_state_lock`。如果工作线程持有同一把锁执行 `os.replace()`，NAS 上较慢的元数据操作会使下一次 `save_config_async()` 在首次 `await` 前等待该锁，从而暂停整个事件循环。

`_save_commit_lock` 只由工作线程获取，用于串行化 `os.replace()`；`_save_state_lock` 仅短暂保护修订号和快照生成。两者职责不同。

该时序由 `test_save_config_async_does_not_block_next_snapshot_during_replace` 覆盖：阻塞第一次 `os.replace()` 后启动第二次异步保存。使用单锁并在锁内替换时测试会因事件循环无法生成下一份快照而失败，当前双锁实现通过。因此维持现有设计。

**Review by Soulter — APPROVED** [review]
